packages/report-server/src/reportBuilder/context/buildOutputContext.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { yup } from '@tupaia/utils';
+import { ReportServerAggregator } from '../../aggregator';
+import { OutputContext } from './types';
+
+type Params = {
+  dataGroups: string[];
+  type: string;
+};
+
+const rawDataExportContextBuilder = async (params: Params, aggregator: ReportServerAggregator) => {
+  const dataElementValidator = yup
+    .array()
+    .of(
+      yup.object().shape({
+        code: yup.string().required(),
+        text: yup.string().required(),
+      }),
+    )
+    .required();
+
+  const { dataGroups } = params;
+
+  const dataElementsMetadata = (
+    await Promise.all(
+      dataGroups.map(async surveyCode => {
+        const { dataElements } = await aggregator.fetchDataGroup(surveyCode);
+        const validatedDataElements = dataElementValidator.validateSync(dataElements);
+
+        return validatedDataElements.map(dataElement => ({
+          key: dataElement.code,
+          title: dataElement.text,
+        }));
+      }),
+    )
+  ).flat();
+  return { dataElementsMetadata };
+};
+
+const outputContextBuilders = {
+  rawDataExport: rawDataExportContextBuilder,
+};
+
+export const buildOutputContext = async (
```

Comment:
Hmmm I'm not too sure we actually want to go with the whole 'context' pattern for the output step. The reason we did it for the transform step is simply that we can't do async logic in the transforms, so we need to compute any async data upfront. 

But we can run the output step as async, so perhaps we just move this logic to get the `dataElementsMetadata` to the `rawDataExportBuilder` instead?